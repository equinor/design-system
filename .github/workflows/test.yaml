name: Testing actions
on:
  push:
    branches:
      - develop
  workflow_dispatch:
    inputs:
      test-value:
        description: 'Test'
        required: false
        default: 'development'
jobs:
  setup-variables:
    uses: equinor/design-system/.github/workflows/setup-env-vars.yml@develop
    with:
      environment: ${{ github.event.inputs.test-value }}
  print-variables:
    runs-on: ubuntu-latest
    needs: [setup-variables]
    steps:
      - name: Workflow environment
        run: echo ${{ needs.setup-variables.outputs.environment }}
      - name: Workflow tag
        run: echo ${{ needs.setup-variables.outputs.tag }}
      - name: Input output
        run: echo ${{ github.event.inputs.test-value }}
