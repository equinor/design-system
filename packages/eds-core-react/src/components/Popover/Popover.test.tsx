/* eslint-disable no-undef */
import { render, cleanup, screen, waitFor, act } from '../../test'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import { Popover } from '.'
import type { PopoverProps } from './Popover'

// We override Popover for testing and set props to partial because AnchorEl is applied to children in custom render function
const TestPopover = Popover as React.ForwardRefExoticComponent<
  Partial<PopoverProps>
>

const StyledPopover = styled(TestPopover)`
  background: red;
`

afterEach(cleanup)
const mockResizeObserver = jest.fn(() => ({
  observe: jest.fn(),
  disconnect: jest.fn(),
  unobserve: jest.fn(),
}))

beforeAll(() => {
  window.ResizeObserver = mockResizeObserver
})

describe('Popover', () => {
  it('Matches snapshot', async () => {
    render(
      <StyledPopover open data-testid="popover">
        <div>some random content</div>
      </StyledPopover>,
    )
    const container = await screen.findByTestId('popover')

    expect(container).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    const { container } = render(
      <TestPopover open>
        <Popover.Title>Title Text</Popover.Title>
        <Popover.Content>Content Text</Popover.Content>
        <Popover.Actions>
          <button type="button">OK</button>
        </Popover.Actions>
      </TestPopover>,
    )
    await act(async () => {
      const results = await axe(container)
      expect(results).toHaveNoViolations()
    })
  })

  it('can extend the css for the component', async () => {
    render(
      <StyledPopover open data-testid="popover">
        <div>some random content</div>
      </StyledPopover>,
    )
    const container = screen.getByTestId('popover')

    await waitFor(() => expect(container).toHaveStyleRule('background', 'red'))
  })
  it('is visible when open is true & anchorEl is set', () => {
    render(
      <TestPopover open placement="right-start" data-testid="popover">
        <div>some random content</div>
      </TestPopover>,
    )
    const container = screen.queryByTestId('popover')
    expect(container).toBeDefined()
    expect(container).toBeVisible()
  })

  it('has rendered Popover Title', async () => {
    render(
      <TestPopover open>
        <Popover.Title>Title Text</Popover.Title>
      </TestPopover>,
    )
    const title = screen.getByText('Title Text')
    await waitFor(() => expect(title).toBeDefined())
  })
  it('has rendered Popover Content', async () => {
    render(
      <TestPopover open>
        <Popover.Content>Content Text</Popover.Content>
      </TestPopover>,
    )
    const content = screen.getByText('Content Text')
    await waitFor(() => expect(content).toBeDefined())
  })
  it("doesn't crash if no children is provided to Popover component", async () => {
    const placement = 'top'
    render(<TestPopover placement={placement} data-testid="popover" open />)
    expect(await screen.findByTestId('popover')).toBeDefined()
  })
  it("doesn't crash if Popover Content children is undefined", async () => {
    const placement = 'top'
    render(
      <TestPopover placement={placement} data-testid="popover" open>
        <Popover.Title>Title</Popover.Title>
        <Popover.Content></Popover.Content>
      </TestPopover>,
    )
    expect(await screen.findByTestId('popover')).toBeDefined()
  })
  it("doesn't crash if Popover Title children is undefined", async () => {
    const placement = 'top'
    render(
      <TestPopover placement={placement} data-testid="popover" open>
        <Popover.Title></Popover.Title>
        <Popover.Content>Content</Popover.Content>
      </TestPopover>,
    )
    expect(await screen.findByTestId('popover')).toBeDefined()
  })
})
