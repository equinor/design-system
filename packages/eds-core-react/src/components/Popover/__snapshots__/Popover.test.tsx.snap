// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover Matches snapshot 1`] = `
.c0 {
  background: var(--eds_ui_background__default, rgba(255, 255, 255, 1));
  box-shadow: 0 1px 10px rgba(0, 0, 0, 0.2),0 4px 5px rgba(0, 0, 0, 0.12),0 2px 4px rgba(0, 0, 0, 0.14);
}

.c1 {
  position: relative;
  margin: 0;
  color: var(--eds_ui__accordion_header_color, rgba(61, 61, 61, 1));
  font-family: Equinor;
  font-size: 1.000rem;
  font-weight: 500;
  line-height: 1.500em;
  text-align: left;
  background: var(--eds_ui_background__default, rgba(255, 255, 255, 1));
  border-radius: 4px;
}

.c1:focus-visible {
  outline: 2px dashed var(--eds_interactive__focus, rgba(0, 112, 121, 1));
  outline-offset: 0px;
}

.c3 {
  position: absolute;
  width: 6px;
  height: 8px;
}

.c5 {
  display: grid;
  grid-gap: 16px;
  max-height: 80vh;
  width: max-content;
  max-width: 560px;
  overflow: auto;
}

.c4 {
  width: 6px;
  height: 8px;
  position: absolute;
  fill: var(--eds_ui_background__default, rgba(255, 255, 255, 1));
  filter: drop-shadow(-4px 0px 2px rgba(0,0,0,0.2));
}

.c2 {
  background: red;
}

<div
  class="c0 c1 c2"
  data-testid="popover"
>
  <div
    class="c3 arrow"
    style="top: -6px; transform: rotate(90deg); left: -1px;"
  >
    <svg
      class="c4 arrowSvg"
    >
      <path
        d="M0.504838 4.86885C-0.168399 4.48524 -0.168399 3.51476 0.504838 3.13115L6 8.59227e-08L6 8L0.504838 4.86885Z"
      />
    </svg>
  </div>
  <div
    class="c5"
  >
    <div>
      some random content
    </div>
  </div>
</div>
`;
